# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/media/webaudio/blink/FFTConvolver.cpp
# Commit: 2356949f5c6d
# Full Hash: 2356949f5c6d9a4f229d012051bac51b3eb83aec
# Author: Paul Adenot <paul@paul.cx>
# Date: 2024-02-28 16:14:38
# Regressor Bug: 1879873
# File Overlap Count: 1
# Description:
#   Bug 1879873 - Scale the inverse FFT result using tx instead of pre- or post-processing the data. r=karlt
#   
#   This skips a copy at the expense of having to do a `const_cast`.
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D202434
# ==============================================================================

diff -r 9b4ea6ecadd1 -r 2356949f5c6d dom/media/webaudio/blink/FFTConvolver.cpp
--- a/dom/media/webaudio/blink/FFTConvolver.cpp	Wed Feb 28 12:50:27 2024 +0000
+++ b/dom/media/webaudio/blink/FFTConvolver.cpp	Wed Feb 28 12:50:27 2024 +0000
@@ -85,7 +85,7 @@
     // The input buffer is now filled (get frequency-domain version)
     m_frame.PerformFFT(m_inputBuffer.Elements());
     m_frame.Multiply(*fftKernel);
-    m_frame.GetInverseWithoutScaling(m_outputBuffer.Elements());
+    m_frame.GetInverse(m_outputBuffer.Elements());
 
     // Overlap-add 1st half from previous time
     AudioBufferAddWithScale(m_lastOverlapBuffer.Elements(), 1.0f,